### PR TITLE
[nest] Avoid registering conflicting filters for SSE connection

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

File: bundles/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/rest/NestStreamingRequestFilter.java [hunk 1]
```diff
@@ -30,18 +33,29 @@
...
 @NonNullByDefault
 public class NestStreamingRequestFilter implements ClientRequestFilter {
-    private final String accessToken;
+    private final Logger logger = LoggerFactory.getLogger(NestStreamingRequestFilter.class);
 
-    public NestStreamingRequestFilter(String accessToken) {
-        this.accessToken = accessToken;
+    private final ConcurrentHashMap<String, String> accessTokens = new ConcurrentHashMap<>();
+
+    public NestStreamingRequestFilter() {
```
Comment (bruestel) [inline bundles/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/rest/NestStreamingRequestFilter.java:40]: Can we remove the default constructor?

Comment (lolodomo) [reply bundles/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/rest/NestStreamingRequestFilter.java]: Removed

**Comment by lolodomo** [comment]
@wborn : can you please confirm that the redirect URL will be different for 2 different Nest accounts (bridges) ? I based on this  to save the access token.

**Comment by wborn** [comment]
No it can also be the same. It depends on to what Nest WWN API server a client is redirected. It can also change over time when a server goes offline.

**Comment by lolodomo** [comment]
> No it can also be the same. It depends on to what Nest WWN API server a client is redirected. It can also change over time when a server goes offline.

In this case,  my fix is not good.

**Comment by wborn** [comment]
I never had any issues with the current implementation when using the Nest Binding with OH 3.0.x. Also it will be difficult for me to test any changes to the WWN implementation since I've switched my account over and can now only use/support the SDM API implementation (https://github.com/openhab/openhab-addons/pull/8947). So unless you know for sure it is severely broken 3.1.x, I'd go with "don't fix what ain't broken", which I don't say often. :wink:

**Comment by lolodomo** [comment]
In fact, there is a potential problem for users setting up two different bridges (2 different Best accounts and so 2 different access tokens). Certainly not something very common and why any issue was never opened.

**Pull request closed without merging**